### ISSUE TITLE
Warnings when compiling documentation

### ISSUE THREAD

**Issue by ExcaliburZero** [typelead/eta#153]
I recently tried compiling the documentation for Eta on the master branch, and the documentation did compile, however it also gave a number of warning messages.

```
[23:01:34] ~/Code/eta/docs master✔
$ make html
sphinx-build -b html -d build/doctrees   source build/html
Running Sphinx v1.5.1
making output directory...
loading pickled environment... not yet created
building [mo]: targets for 0 po files that are out of date
building [html]: targets for 5 source files that are out of date
updating environment: 5 added, 0 changed, 0 removed
reading sources... [100%] stack-traces                                                        
/home/chris/Code/eta/docs/source/faq.rst:15: WARNING: Title underline too short.

Is Eta ready to use?
-------------------
/home/chris/Code/eta/docs/source/faq.rst:15: WARNING: Title underline too short.

Is Eta ready to use?
-------------------
/home/chris/Code/eta/docs/source/faq.rst:19: WARNING: Title underline too short.

When will Eta make its first official release?
------------------------------------------
/home/chris/Code/eta/docs/source/faq.rst:19: WARNING: Title underline too short.

When will Eta make its first official release?
------------------------------------------
/home/chris/Code/eta/docs/source/faq.rst:29: WARNING: Title underline too short.

Will Eta have an interpreter and support TemplateHaskell in the future?
--------------------------------------------------------------------
/home/chris/Code/eta/docs/source/faq.rst:29: WARNING: Title underline too short.

Will Eta have an interpreter and support TemplateHaskell in the future?
--------------------------------------------------------------------
/home/chris/Code/eta/docs/source/faq.rst:33: WARNING: Title underline too short.

Will Eta be compatible with GHC 8?
---------------------------------
/home/chris/Code/eta/docs/source/faq.rst:33: WARNING: Title underline too short.

Will Eta be compatible with GHC 8?
---------------------------------
/home/chris/Code/eta/docs/source/faq.rst:44: WARNING: Title underline too short.

Why not make Eta a part of GHC? What will be different about the two?
-------------------------------------------------------------------
/home/chris/Code/eta/docs/source/faq.rst:44: WARNING: Title underline too short.

Why not make Eta a part of GHC? What will be different about the two?
-------------------------------------------------------------------
/home/chris/Code/eta/docs/source/faq.rst:60: WARNING: Title underline too short.

Why not work along with Haskell.org and help them improve their infrastructure and tooling?
--------------------------------------------------------------------------------------
/home/chris/Code/eta/docs/source/faq.rst:60: WARNING: Title underline too short.

Why not work along with Haskell.org and help them improve their infrastructure and tooling?
--------------------------------------------------------------------------------------
/home/chris/Code/eta/docs/source/faq.rst:71: WARNING: Title underline too short.

Is Eta very different from GHC 7.10.3 in its initial state?
----------------------------------------------------
/home/chris/Code/eta/docs/source/faq.rst:71: WARNING: Title underline too short.

Is Eta very different from GHC 7.10.3 in its initial state?
----------------------------------------------------
/home/chris/Code/eta/docs/source/faq.rst:79: WARNING: Title underline too short.

How is Eta different from Frege?
------------------------------
/home/chris/Code/eta/docs/source/faq.rst:79: WARNING: Title underline too short.

How is Eta different from Frege?
------------------------------
/home/chris/Code/eta/docs/source/ffi.rst:2: WARNING: Title underline too short.

Interacting with Java
==================
/home/chris/Code/eta/docs/source/ffi.rst:20: WARNING: Title underline too short.

Unboxed & Primitive Types
^^^^^^^^^^^^^^^^^^^^^^^^
/home/chris/Code/eta/docs/source/ffi.rst:44: WARNING: Title underline too short.

Primitive Types Reference
^^^^^^^^^^^^^^^^^^^^^^^
/home/chris/Code/eta/docs/source/ffi.rst:44: WARNING: Title underline too short.

Primitive Types Reference
^^^^^^^^^^^^^^^^^^^^^^^
/home/chris/Code/eta/docs/source/ffi.rst:148: ERROR: Unexpected indentation.
/home/chris/Code/eta/docs/source/ffi.rst:149: WARNING: Block quote ends without a blank line; unexpected unindent.
/home/chris/Code/eta/docs/source/ffi.rst:197: WARNING: Title underline too short.

Foreign imports
^^^^^^^^^^^^^^
/home/chris/Code/eta/docs/source/ffi.rst:201: WARNING: Title underline too short.

Foreign exports
^^^^^^^^^^^^^^
/home/chris/Code/eta/docs/source/ffi.rst:201: WARNING: Title underline too short.

Foreign exports
^^^^^^^^^^^^^^
/home/chris/Code/eta/docs/source/ffi.rst:210: ERROR: Unexpected indentation.
/home/chris/Code/eta/docs/source/ffi.rst:211: WARNING: Block quote ends without a blank line; unexpected unindent.
/home/chris/Code/eta/docs/source/getting-started.rst:2: WARNING: Title underline too short.

Getting Started with Eta
======================
/home/chris/Code/eta/docs/source/stack-traces.rst:122: WARNING: Enumerated list ends without a blank line; unexpected unindent.
/home/chris/Code/eta/docs/source/stack-traces.rst:128: WARNING: Definition list ends without a blank line; unexpected unindent.
looking for now-outdated files... none found
pickling environment... done
checking consistency... done
preparing documents... done
/home/chris/Code/eta/docs/source/stack-traces.rst:103: WARNING: Could not lex literal_block as "java". Highlighting skipped.
/home/chris/Code/eta/docs/source/stack-traces.rst:110: WARNING: Could not lex literal_block as "java". Highlighting skipped.

generating indices... genindex
writing additional pages... search
copying static files... WARNING: html_static_path entry u'/home/chris/Code/eta/docs/source/_static' does not exist
done
copying extra files... done
dumping search index in English (code: en) ... done
dumping object inventory... done
build succeeded, 33 warnings.

Build finished. The HTML pages are in build/html.
```

**Comment by ExcaliburZero** [comment]
I have made a pull request #154, which fixes all of these warnings except for the last one, which I am unsure how to fix.

**Comment by ExcaliburZero** [comment]
I compiled the documentation after 30402c3fb62e12f7a78433084bc668a419700443, and just the last warning is left.

```
[0:18:47] ~/Code/eta/docs master✔
 $ make html
sphinx-build -b html -d build/doctrees   source build/html
Running Sphinx v1.5.1
making output directory...
loading pickled environment... not yet created
building [mo]: targets for 0 po files that are out of date
building [html]: targets for 5 source files that are out of date
updating environment: 5 added, 0 changed, 0 removed
reading sources... [100%] stack-traces                                                        
looking for now-outdated files... none found
pickling environment... done
checking consistency... done
preparing documents... done
writing output... [100%] stack-traces                                                         
generating indices... genindex
writing additional pages... search
copying static files... WARNING: html_static_path entry u'/home/chris/Code/eta/docs/source/_static' does not exist
done
copying extra files... done
dumping search index in English (code: en) ... done
dumping object inventory... done
build succeeded, 1 warning.

Build finished. The HTML pages are in build/html.
```

**Comment by rahulmutt** [comment]
Fixed in 9e11f39  - git doesn't track empty directories so I added a stub file inside of `_static` so that it does.

**Comment by ExcaliburZero** [comment]
Can confirm that it now compiles without any warnings.
